src/detector/internals.rs

```diff
@@ -93,21 +88,22 @@ pub fn pitch_from_peaks<T>(
 where
     T: Float,
 {
+    let sample_rate = T::from_usize(sample_rate).unwrap();
     let peaks = detect_peaks(input);
+
     choose_peak(peaks, clarity_threshold)
         .map(|peak| correct_peak(peak, input, correction))
-        .map(|peak| {
-            let frequency = T::from_usize(sample_rate).unwrap() / peak.0;
-            let clarity = peak.1 / input[0];
-            Pitch { frequency, clarity }
+        .map(|peak| Pitch {
+            frequency: sample_rate / peak.0,
+            clarity: peak.1 / input[0],
         })
 }
 
 pub fn get_power_level<T>(signal: &[T]) -> T
 where
     T: Float + std::iter::Sum,
 {
-    signal.iter().map(|s| *s * *s).sum::<T>()
+    signal.iter().map(|&s| s * s).sum::<T>()
```

Comment:
It turns out a change like this can have potentially big performance consequences....I will look into benchmarking these.

Reply:
Thanks for noticing it, I was about to merge it without giving it a second thought.

Let me know what you find out from the benchmarks

Reply:
No problem. Like I said, I'm a beginner...but, I will push a few more updates soon. I refactors the peak finder and it is 3x faster now :-).

Reply:
I've benchmarked, and it appears not to make a measureable difference. I think it looks cleaner with the extra `*`s, so I'll leave it like this if you don't object.